whylog/tests/tests_config/test_get_search_range.py

```diff
@@ -20,22 +22,70 @@ def setUpClass(cls):
         cause1_regex = '^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) cause1 transaction number: (\d+) Host: (\w)$'
         cause1_line = '2016-04-12 23:39:43 cause1 transaction number: 10101 Host: db_host'
         convertions = {1: 'date'}
-        cause1 = RegexParser("cause1", cause1_line, cause1_regex, [1], 'database', convertions)
+        cls.cause1 = RegexParser("cause1", cause1_line, cause1_regex, [1], 'database', convertions)
 
         cause2_regex = '^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) cause2 moved resource id: (\d+) Host: (\w)$'
         cause2_line = '2016-04-12 23:40:43 cause2 moved resource id: 1234 Host: apache_host'
         convertions = {1: 'date'}
-        cause2 = RegexParser("cause2", cause2_line, cause2_regex, [1], 'apache', convertions)
+        cls.cause2 = RegexParser("cause2", cause2_line, cause2_regex, [1], 'apache', convertions)
 
         effect_regex = '^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) effect internal server error Host: (\w)$'
         effect_line = '2016-04-12 23:54:43 effect internal server error Host: apache_host'
         convertions = {1: 'date'}
-        effect = RegexParser("effect", effect_line, effect_regex, [1], 'apache', convertions)
+        cls.effect = RegexParser("effect", effect_line, effect_regex, [1], 'apache', convertions)
+
+        line_source = LineSource('localhost', 'node_1.log')
+        cls.effect_time = datetime(2016, 4, 12, 23, 54, 43)
+        effect_line = '2016-04-12 23:54:43 effect internal server error Host: apache_host'
+        cls.effect_clues = {'effect': Clue((cls.effect_time,), effect_line, 40, line_source)}
+
+        cls.earliest_date = datetime(1, 1, 1, 1, 1, 1)
+        cls.ten_second_earlier = datetime(2016, 4, 12, 23, 54, 33)
+        cls.one_hundred_second_earlier = datetime(2016, 4, 12, 23, 53, 3)
+        cls.ten_second_later = datetime(2016, 4, 12, 23, 54, 53)
+
+    @classmethod
+    def calculate_range(cls, rule):
+        cls.config._rules['apache'].append(rule)
+        calculated_ranges = cls.config._get_search_ranges([rule], cls.effect_clues)
+        cls.config._rules['apache'].pop()
```

Comment:
I'm afraid I don't understand role of config in this function. Append and pop to _rules seems to be useless.